### ISSUE TITLE
Failed to build demo projects

### ISSUE THREAD

**Issue by nitosan** [gearvrf/GearVRf-Demos#569]
Failed to build with command
`.\gradlew assembleDebug`

### Error Stack
```
GearVRf-Demos\gvr-3dcursor-simple\app\src\main\java\org\gearvrf\io\cursorsimple\CursorMain.java:121: error: <anonymous org.gearvrf.io.cursorsimple.CursorMain$1> is not abstract and does not override abstract method onStateChanged(SelectableBehavior,ObjectState,ObjectState,Cursor) in StateChangedListener
        movableCubeBehavior.setStateChangedListener(new StateChangedListener() {
                                                                               ^
GearVRf-Demos\gvr-3dcursor-simple\app\src\main\java\org\gearvrf\io\cursorsimple\CursorMain.java:122: error: method does not override or implement a method from a supertype
            @Override
            ^
GearVRf-Demos\gvr-3dcursor-simple\app\src\main\java\org\gearvrf\io\cursorsimple\CursorMain.java:189: error: <anonymous org.gearvrf.io.cursorsimple.CursorMain$2> is not abstract and does not override abstract method onStateChanged(SelectableBehavior,ObjectState,ObjectState,Cursor) in StateChangedListener
        movableBehavior.setStateChangedListener(new StateChangedListener() {
                                                                           ^
GearVRf-Demos\gvr-3dcursor-simple\app\src\main\java\org\gearvrf\io\cursorsimple\CursorMain.java:190: error: method does not override or implement a method from a supertype
            @Override
            ^
4 errors
:gvr-3dcursor-simple:compileDebugJavaWithJavac FAILED
```

**Comment by thomasflynn** [comment]
yeah, we're working on it.  I think we found a root cause.  Abhijit is testing a fix.

**Comment by nitosan** [comment]
I was able to build demos with local GVRF. 
But I think for developers the demo should work out of the box, so will keep this issue open for now.

**Comment by apjagdale** [comment]
@nitosan But we have release build for that. Current demos (master branch) will build only with current framework.

**Comment by thomasflynn** [comment]
yeah, we're close to fixing it.  There were several problems in the build script for starters.  We've sorted through most of those and then ran into a bug in the gradle file for gvr-bullet.  Abhijit's pull-request fixes that gradle file.  After that, there is one more fix needed for the build script.  That's in testing now.  Hopefully, that'll work and we can re-trigger the build for the current pull-requests.

**Comment by nitosan** [comment]
@apjagdale can we create a dev branch for work-in-progress code? Usually master branch means stable code.

Because whenever a developer clone/fork our code, they will get our master branch (because that's the default branch). If it failed to compile, it will leave a bad impression. 

If we use a separate branch such as 'dev', the developer won't be effected.

**Comment by thomasflynn** [comment]
Ok, i misunderstood what you were asking.  There is a pre-merge build problem that we are sorting out that is holding up some check-ins.  That's what I was replying about.

The development model here internally and with most projects we've worked on use 'master' as the dev branch and then, when stable, do a release.  That release is tagged and branched.  Should any minor fixes be needed between releases, that minor fix is done to that branch (e.g. when we did a 3.0.1 release a long time back.  we were in the midst of API changes, so made the minor fix on the 3.0 branch and released the update as 3.0.1).

So in other words, 'master' of GearVRf-Demos should match up with 'master' of GearVRf.  Ditto with the 4.0 branch of each.

That being said, the Demos are being built against a 4.0.1-SNAPSHOT we did immediately after the 4.0 release.  We should do a 4.1 snapshot and update the demos to that snapshot.  We should also regularly update the snapshot.

**Comment by nitosan** [comment]
@thomasflynn Is it possible that we create a `stable` branch which always point to the latest release? And use that as the default branch?

I want to make sure the developer that clone this repo will always have code that works out of the box.

**Comment by thomasflynn** [comment]
Right, as I mentioned, the SNAPSHOT should be updated.  Actually, should be updated every time we change the API.  As long as we do that, then anytime someone clones the demos repo, it should build out of the box.

**Comment by thomasflynn** [comment]
i believe the snapshot has been updated.  close this?
